### ISSUE TITLE
Support for `NativeEnvironment` instead of rendering to String

### ISSUE THREAD

**Issue by noxasaxon** [mitsuhiko/minijinja#50]
Hi @mitsuhiko, 
I was wondering if you think its possible for this library to support jinja2's [`NativeEnvironment` feature](https://github.com/pallets/jinja/blob/417f822196f66155e8c121e5229cc12a6b02ce14/docs/nativetypes.rst), to parse code into a type other than string. 

I feel like there's some magic involved in serde_json that might make this do-able, but I am also too new with rust to know whether looking into this would be a dead end 😆

**Comment by mitsuhiko** [comment]
What would you want to use this for? It can probably be done but supporting this type of feature is relatively complex.

**Comment by noxasaxon** [comment]
**TLDR:** Users can define jinja templates as input to a parameter for a Lambda Function, and our library will resolve that template to any type (not just String) before handing off the resolved parameter to their lambda's business logic.

Do you have an idea of where I should start poking around in minijinja to possibly make a PR, and what hangups to look out for?
--- 
**Longer answer:**
I'm experimenting with porting [this library](https://github.com/twilio-labs/socless_python) that enhances AWS Step Functions to [Rust](https://github.com/noxasaxon/socless_rust) so we have the option to use Rust lambdas when necessary.

One of the key enhancements to Step Functions is that we provide a Global State, and jinja templating to reference things from global state when you send them to a specific lambda step (allowing things like list comprehensions without writing a specific lambda just for that use case). Because we are handing off resolved templates to a python function instead of HTML, it needs to resolve to any possible JSON type like `int`, `array`, etc.

**Comment by mitsuhiko** [comment]
Generally speaking I'm in favor of this but this goes beyond what I using myself. As you might be aware I'm generally using Jinja for expression expansion which is why `Expression` already exists. If you are willing to implement this behind a feature flag I'm very much in favor of having it but I am not using this use case myself so it's hard to provide myself.

**Comment by mitsuhiko** [comment]
I think the combination of #82 and the custom formatter support there is enough in the engine now that supports this use case.
